File: php/features.php
<?php
declare(strict_types=1);

// PlacarAgora SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PlacarAgoraFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PlacarAgoraBaseFeature();
            case "test":
                return new PlacarAgoraTestFeature();
            default:
                return new PlacarAgoraBaseFeature();
        }
    }
}

File: py/features.py
# PlacarAgora SDK feature factory

from feature.base_feature import PlacarAgoraBaseFeature
from feature.test_feature import PlacarAgoraTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PlacarAgoraBaseFeature(),
        "test": lambda: PlacarAgoraTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
